### PR TITLE
Fork the lockdown test to two tests as it is doing the work of 2

### PR DESCRIPTION
**Pull request by kfox1111**
This patch makes a second copy of the existing lockdown test
because it is doing double duty. In follow on patches we will
make each test do one thing only.

### CODE REVIEW

File: .github/tests/namespace-override/pre-install.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+kubectl create namespace "spire-system"
+kubectl label namespace "spire-system" pod-security.kubernetes.io/enforce=privileged
+kubectl create namespace "spire-server"
+kubectl label namespace "spire-server" pod-security.kubernetes.io/enforce=restricted
```
Comment (marcofranssen) [inline .github/tests/namespace-override/pre-install.sh:5]:
```suggestion
kubectl create namespace spire-system
kubectl label namespace spire-system pod-security.kubernetes.io/enforce=privileged
kubectl create namespace spire-server
kubectl label namespace spire-server pod-security.kubernetes.io/enforce=restricted
```

File: .github/tests/namespace-override/post-install.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+k_rollout_status=(kubectl rollout status --watch --timeout 30s --namespace)
+
+cat <<EOF >>"$GITHUB_STEP_SUMMARY"
+### spire
+| workload | Status |
+| -------- | ------ |
+| spire-server | $("${k_rollout_status[@]}" spire-server statefulset spire-server) |
+| spire-spiffe-csi-driver | $("${k_rollout_status[@]}" spire-system daemonset spire-spiffe-csi-driver) |
+| spire-agent | $("${k_rollout_status[@]}" spire-system daemonset spire-agent) |
+| spire-spiffe-oidc-discovery-provider | $("${k_wait[@]}" spire-server deployments.apps spire-spiffe-oidc-discovery-provider) |
```
Comment (marcofranssen) [inline .github/tests/namespace-override/post-install.sh:18]:
```suggestion
| spire-server | $("${k_rollout_status[@]}" spire-server statefulset spire-server) |
| spire-spiffe-oidc-discovery-provider | $("${k_wait[@]}" spire-server deployments.apps spire-spiffe-oidc-discovery-provider) |
| spire-spiffe-csi-driver | $("${k_rollout_status[@]}" spire-system daemonset spire-spiffe-csi-driver) |
| spire-agent | $("${k_rollout_status[@]}" spire-system daemonset spire-agent) |
```

**Review by marcofranssen — CHANGES_REQUESTED** [review]
Could you add a temporary commit to run the tests? Also few nit-picky suggestions in there.

File: .github/tests/namespace-override/pre-install.sh (hunk 1 above)
Comment (kfox1111) [reply .github/tests/namespace-override/pre-install.sh]: Done

**Comment by kfox1111** [comment]
This is ready to go, but needs the version bump reverted. It was put in place to ensure all the changes work.

File: .github/tests/namespace-override/pre-install.sh [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
```
Comment (marcofranssen) [inline .github/tests/namespace-override/pre-install.sh:1]:
```suggestion
#!/usr/bin/env bash

```

File: .github/tests/namespace-override/.env [hunk 4]
```diff
@@ -0,0 +1 @@
+export EXTRA_HELM_ARGS="--values=examples/production/values.yaml"
```
Comment (marcofranssen) [inline .github/tests/namespace-override/.env:1]: This should not be part of this test. I would expect just the namespace override.

**Review by marcofranssen — DISMISSED** [review]
Can we rename this PR to just

`Replace lockdown test with namespace-override test`?

#127 replaces the lockdown test with just testing the production example (which is already locked down).

I even debate the value of testing these 2 namespace overrides. Feels a bit like this PR is obsolete when we merge #127

**Comment by kfox1111** [comment]
Trying to maintain history. the existing lockdown test is doing 2 things. testing production values, and namespace-override testing. To follow the established principal of one pr doing just one thing, we need to fork the test into the two so its obvious where the test came from. We then can continue to enhance both test separately in separate pr's.

**Review by faisal-memon — APPROVED** [review]

**Pull request merged**